plugins/obs-filters/luma-key-filter.c

```diff
@@ -0,0 +1,187 @@
+#include <obs-module.h>
+
+#define SETTING_LUMA_MAX           "luma_max"
+#define SETTING_LUMA_MIN           "luma_min"
+#define SETTING_LUMA_MAX_SMOOTH    "luma_max_smooth"
+#define SETTING_LUMA_MIN_SMOOTH    "luma_min_smooth"
+#define SETTING_COLOR		   "color"
+#define SETTING_INVERT_LUMA	   "invert_luma"
+#define SETTING_INVERT_COLOR	   "invert_color"
+
+#define TEXT_LUMA_MAX           obs_module_text("Luma.LumaMax")
+#define TEXT_LUMA_MIN           obs_module_text("Luma.LumaMin")
+#define TEXT_LUMA_MAX_SMOOTH    obs_module_text("Luma.LumaMaxSmooth")
+#define TEXT_LUMA_MIN_SMOOTH    obs_module_text("Luma.LumaMinSmooth")
+#define TEXT_COLOR              obs_module_text("Color")
+#define TEXT_INVERT_LUMA	obs_module_text("Invert Luma")
+#define TEXT_INVERT_COLOR	obs_module_text("Invert Color")
+
+
+struct luma_key_filter_data {
+	obs_source_t    *context;
+
+	gs_effect_t     *effect;
+
+	gs_eparam_t     *luma_max_param;
+	gs_eparam_t     *luma_min_param;
+	gs_eparam_t     *luma_max_smooth_param;
+	gs_eparam_t     *luma_min_smooth_param;
+	gs_eparam_t     *color_param;
+	gs_eparam_t     *invert_color_param;
+	gs_eparam_t     *invert_luma_param;
+
+
+	float           luma_max;
+	float           luma_min;
+	float           luma_max_smooth;
+	float           luma_min_smooth;
+	struct vec4     color;
+	bool		invert_color;
+	bool		invert_luma;
+};
+
+static const char *luma_key_name(void *unused)
+{
+	UNUSED_PARAMETER(unused);
+	return obs_module_text("LumaKeyFilter");
+}
+
+static void luma_key_update(void *data, obs_data_t *settings)
+{
+	struct luma_key_filter_data *filter = data;
+
+	double lumaMax = obs_data_get_double(settings, SETTING_LUMA_MAX);
+	double lumaMin = obs_data_get_double(settings, SETTING_LUMA_MIN);
+	double lumaMaxSmooth = obs_data_get_double(settings, SETTING_LUMA_MAX_SMOOTH);
```

Comment:
Style:  Line exceeds 80 characters.

```suggestion
	double lumaMaxSmooth = obs_data_get_double(settings,
			SETTING_LUMA_MAX_SMOOTH);
```